pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -988,17 +988,23 @@ func translateHeaderMatch(name string, in *networking.StringMatch) *route.Header
 
 	switch m := in.MatchType.(type) {
 	case *networking.StringMatch_Exact:
-		out.HeaderMatchSpecifier = &route.HeaderMatcher_ExactMatch{ExactMatch: m.Exact}
+		if m.Exact != "" {
```

Comment:
this doesn't seem to solve it, just makes it not do anything.

https://github.com/istio/istio/blob/90e61d212dce2550e29250afb053ed45099adb6a/pilot/pkg/networking/core/v1alpha3/route/route.go#L923 is the code intended to solve this I believe. I think its incorrectly checking `nil` when `""` can occur as well?

Reply:
Although I think maybe this should be invalid and rejected in the webhook.

Reply:
> this doesn't seem to solve it, just makes it not do anything.
> 
> https://github.com/istio/istio/blob/90e61d212dce2550e29250afb053ed45099adb6a/pilot/pkg/networking/core/v1alpha3/route/route.go#L923
> 
> is the code intended to solve this I believe. I think its incorrectly checking `nil` when `""` can occur as well?

  In the absence of any header match specifier, match will default to present_match, so it seems work in envoy when you set  prefix: ""  
headers:
    x-my-header:
      prefix: "" # Empty value means only the presence of the header is checked

==>
          "match": {
           "path": "/productpage",
           "case_sensitive": true,
           "headers": [
            {
             "name": "x-my-header"
            }
           ]
          }

Reply:
according `(repeated config.route.v3.HeaderMatcher) Specifies a set of headers that the route should match on. The router will check the request’s headers against all the specified headers in the route config. A match will happen if all the headers in the route are present in the request with the same values (or based on presence if the value field is not in the config)`  ==>  so  if headers only contain name, the match will default to present_match , it's envoy feature

Reply:
@cywang1905 hello

Reply:
= =